src/k8s-engine/progressTracker.ts

```diff
@@ -11,6 +19,29 @@ export default class ProgressTracker {
     this.notify = notify;
   }
 
+  /**
+   * A function that notifies whatever needs to be notified of
+   * changes to the state of progress.
+   */
+  protected notify: (progress: K8s.KubernetesProgress) => void;
+
+  /**
+   * A progress object that is preferred over progress objects that
+   * correspond to actions when passing one to .notify.
```

Comment:
Might be useful to describe this as having, essentially, infinitely high priority.